PSReadLine/History.cs

```diff
@@ -146,7 +171,7 @@ private string GetHistorySaveFileMutexName()
         {
             // Return a reasonably unique name - it's not too important as there will rarely
             // be any contention.
-            return "PSReadLineHistoryFile_" + _options.HistorySavePath.GetHashCode();
+            return "PSReadLineHistoryFile_" + FNV1a32Hash.ComputeHash(_options.HistorySavePath);
```

Comment:
I was going to suggest avoiding the conversion to byte array (can we use a span somehow?) but then realized we probably should be normalizing the name anyway, e.g. ensuring consistent case on Windows.

Reply:
> we probably should be normalizing the name anyway

Agreed. Update `Set-PSReadLineOption` to sort of resolve (not really) PSPath. Also call `ToLower` before sending the file path to hash method.

> avoiding the conversion to byte array (can we use a span somehow?)

This is a fun exercise!
We cannot use span because we need to compile it against .NET 4.6.1, but we don't really need to use a span here.
The update `ComputeHash(string input)` behaves exactly the same as the `ComputeHash(string input)` implementation in the following code
```
internal static uint ComputeHash(byte[] buffer)
{
    uint hash = FNV32_OFFSETBASIS;
    for (int i = 0; i < buffer.Length; i++)
    {
        hash = (hash ^ buffer[i]) * FNV32_PRIME;
    }
    return hash;
}

internal static uint ComputeHash(string input)
{
    return ComputeHash(Encoding.Unicode.GetBytes(input));
}
```